File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        int n = nums.length;

        if (n == 1 && nums[0] == 0) return "0";

        String[] numsStr = new String[n];

        for (int i = 0; i < n; i++) {
            numsStr[i] = String.valueOf(nums[i]);
        }

        Arrays.sort(numsStr, (a, b) -> (b + a).compareTo(a + b));

        StringBuilder sb = new StringBuilder();
        for (String s : numsStr) sb.append(s);

        String result = sb.toString();

        return result.charAt(0) == '0' ? "0" : result;
    }
}

File: 0179-largest-number/0179-largest-number.kt
class Solution {
    fun largestNumber(nums: IntArray): String {
        if (nums.size == 1 && nums[0] == 0) return "0"
        
        val result = nums
            .map { it.toString() }
            .sortedWith { a, b -> (b + a).compareTo(a + b) }
            .joinToString("")
        
        return if (result.startsWith("0")) "0" else result
    }
}